refactor: convert `module` term to `block` [BD-13]

0x29a opened the pull request:
## Description

~~**Note**: for now, the PR depends on and includes https://github.com/openedx/edx-platform/pull/31384, so use [this link](https://github.com/openedx/edx-platform/pull/31475/files/488ae04b54b47bb93df42f64c651e6e31477fe60..9e26985747b2adc507a2323ad6140fd77d3ab81f) to see and review the actual diff.~~

This PR is the continuation of our [previous effort](https://github.com/openedx/edx-platform/pull/31113) to unify XBlock naming across the platform. This time, each `module` occurrence reviewed and changed to `block` if it satisfied these criteria:
- It meant "XBlock".
- It was in `*.py`, `*.rst`, or `*.md` file.
- It didn't mean "Module Store" / "Module System" / "Student Module".
- It could be renamed without causing more confusion than clarity. For example, I wasn't renaming `module` to `block` alongside `module_id`, which can't be renamed as it's the part of the CCX API. Same about `module_data` and `module_state_key`.

This regex has been used to filter out the larger part of false-positive occurrences:
```regex
(?i)(?![^\s]*?store[^\s]*)(?![^\s]*?xmodule\.[^\s]*)(\b[^\s]*?module[^\s]*\b)
``` 
Basically, it excludes `modulestore` and `from xmodule.<something> import <something>` from the search results.

The following table shows the distribution by files:

| **Level 1** | **Level 2**  | **Level 3**                      | **Occurrences** | Notes                                                                                                                                                                                          |
| ----------- | ------------ | -------------------------------- | --------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| openedx     |              |                                  | 617             |                                                                                                                                                                                                |
| openedx     | core         |                                  | 564             |                                                                                                                                                                                                |
| openedx     | core         | djangolib                        | 5               |                                                                                                                                                                                                |
| openedx     | core         | djangoapps                       | 356             |                                                                                                                                                                                                |
| openedx     | core         | lib                              | 200             |                                                                                                                                                                                                |
| openedx     | features     |                                  | 31              |                                                                                                                                                                                                |
| openedx     | features     | course\_experience               | 16              |                                                                                                                                                                                                |
| openedx     | features     | enterprise\_support              | 3               |                                                                                                                                                                                                |
| openedx     | features     | content\_type\_gating            | 5               |                                                                                                                                                                                                |
| openedx     | features     | discounts                        | 1               |                                                                                                                                                                                                |
| openedx     | features     | calendar\_sync                   | 3               |                                                                                                                                                                                                |
| openedx     | features     | personalized\_learner\_schedules | 3               |                                                                                                                                                                                                |
| openedx     | tests        |                                  | 22              |                                                                                                                                                                                                |
| openedx     | tests        | completion\_integration          | 20              |                                                                                                                                                                                                |
| openedx     | tests        | ci                               | 2               |                                                                                                                                                                                                |
| cms         |              |                                  | 366             | Also renamed `item` -> `block`, including `cms/djangoapps/contentstore/views/item.py` file.                                                                                                    |
| cms         | djangoapps   |                                  | 316             |                                                                                                                                                                                                |
| cms         | djangoapps   | xblock\_config                   | 4               |                                                                                                                                                                                                |
| cms         | djangoapps   | pipeline\_js                     | 4               |                                                                                                                                                                                                |
| cms         | djangoapps   | contentstore                     | 305             |                                                                                                                                                                                                |
| cms         | djangoapps   | models                           | 1               |                                                                                                                                                                                                |
| cms         | djangoapps   | api                              | 2               |                                                                                                                                                                                                |
| cms         | lib          |                                  | 1               |                                                                                                                                                                                                |
| cms         | lib          | xblock                           | 1               |                                                                                                                                                                                                |
| cms         | envs         |                                  | 32              |                                                                                                                                                                                                |
| xmodule     |              |                                  | 2206            | Didn't rename xblocks fields, didn't touch modulestore API.                                                                                                                                    |
| xmodule     | capa         |                                  | 92              |                                                                                                                                                                                                |
| xmodule     | capa         | safe\_exec                       | 54              |                                                                                                                                                                                                |
| xmodule     | capa         | tests                            | 24              |                                                                                                                                                                                                |
| xmodule     | modulestore  |                                  | 654             | Didn't rename anything meaning "module store".                                                                                                                                                 |
| xmodule     | modulestore  | perf\_tests                      | 3               |                                                                                                                                                                                                |
| xmodule     | modulestore  | split\_mongo                     | 81              |                                                                                                                                                                                                |
| xmodule     | modulestore  | mongo                            | 109             |                                                                                                                                                                                                |
| xmodule     | modulestore  | tests                            | 158             |                                                                                                                                                                                                |
| xmodule     | contentstore |                                  | 8               |                                                                                                                                                                                                |
| xmodule     | docs         |                                  | 1               |                                                                                                                                                                                                |
| xmodule     | docs         | decisions                        | 1               |                                                                                                                                                                                                |
| xmodule     | util         |                                  | 4               |                                                                                                                                                                                                |
| xmodule     | tests        |                                  | 832             |                                                                                                                                                                                                |
| xmodule     | tests        | rendering                        | 3               |                                                                                                                                                                                                |
| xmodule     | tests        | xml                              | 15              |                                                                                                                                                                                                |
| xmodule     | video\_block |                                  | 34              |                                                                                                                                                                                                |
| scripts     |              |                                  | 48              |                                                                                                                                                                                                |
| scripts     | xsslint      |                                  | 19              |                                                                                                                                                                                                |
| scripts     | xsslint      | xsslint                          | 12              |                                                                                                                                                                                                |
| scripts     | xsslint      | tests                            | 7               |                                                                                                                                                                                                |
| scripts     | xblock       |                                  | 1               |                                                                                                                                                                                                |
| common      |              |                                  | 201             |                                                                                                                                                                                                |
| common      | test         |                                  | 13              |                                                                                                                                                                                                |
| common      | test         | data                             | 3               |                                                                                                                                                                                                |
| common      | djangoapps   |                                  | 188             |                                                                                                                                                                                                |
| common      | djangoapps   | entitlements                     | 2               |                                                                                                                                                                                                |
| common      | djangoapps   | course\_modes                    | 4               |                                                                                                                                                                                                |
| common      | djangoapps   | static\_replace                  | 2               |                                                                                                                                                                                                |
| common      | djangoapps   | track                            | 31              |                                                                                                                                                                                                |
| common      | djangoapps   | student                          | 20              |                                                                                                                                                                                                |
| common      | djangoapps   | split\_modulestore\_django       | 1               |                                                                                                                                                                                                |
| common      | djangoapps   | xblock\_django                   | 3               |                                                                                                                                                                                                |
| common      | djangoapps   | util                             | 64              |                                                                                                                                                                                                |
| common      | djangoapps   | edxmako                          | 26              |                                                                                                                                                                                                |
| common      | djangoapps   | pipeline\_mako                   | 7               |                                                                                                                                                                                                |
| common      | djangoapps   | third\_party\_auth               | 28              |                                                                                                                                                                                                |
| docs        |              |                                  | 148             |                                                                                                                                                                                                |
| docs        | decisions    |                                  | 41              |                                                                                                                                                                                                |
| docs        | guides       |                                  | 103             |                                                                                                                                                                                                |
| docs        | guides       | hooks                            | 2               |                                                                                                                                                                                                |
| docs        | guides       | testing                          | 18              |                                                                                                                                                                                                |
| docs        | guides       | frontend                         | 24              |                                                                                                                                                                                                |
| docs        | guides       | docstrings                       | 32              |                                                                                                                                                                                                |
| docs        | technical    |                                  | 1               |                                                                                                                                                                                                |
| docs        | api          |                                  | 3               |                                                                                                                                                                                                |
| lms         |              |                                  | 2435            | Didn't rename `module_score` and `module_completion`, as they're used together, and `module_score` is the LTI consumer block field. Didn't touch `module_id` because of the same reason.       |
| lms         | djangoapps   |                                  | 2355            |                                                                                                                                                                                                |
| lms         | djangoapps   | mobile\_api                      | 79              |                                                                                                                                                                                                |
| lms         | djangoapps   | verify\_student                  | 15              |                                                                                                                                                                                                |
| lms         | djangoapps   | bulk\_email                      | 5               |                                                                                                                                                                                                |
| lms         | djangoapps   | ora\_staff\_grader               | 2               |                                                                                                                                                                                                |
| lms         | djangoapps   | discussion                       | 69              |                                                                                                                                                                                                |
| lms         | djangoapps   | static\_template\_view           | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | learner\_dashboard               | 12              |                                                                                                                                                                                                |
| lms         | djangoapps   | program\_enrollments             | 2               |                                                                                                                                                                                                |
| lms         | djangoapps   | courseware                       | 1011            | Didn't rename 1) `StudentModule` and anything related, as this would generate Django migration. 2) `StudentModule`'s fields. Also, removed unused `_has_access_xmodule` from `access.py` file. | 
| lms         | djangoapps   | instructor\_task                 | 396             | Didn't rename anything related to `module state`, or `XModule` in in the context of `StudentModule`.                                                                                           |
| lms         | djangoapps   | lms\_xblock                      | 18              |                                                                                                                                                                                                |
| lms         | djangoapps   | ccx                              | 131             | Didn't rename anything in this directory, because `module` term is extensively used in the REST API.                                                                                           |
| lms         | djangoapps   | teams                            | 2               |                                                                                                                                                                                                |
| lms         | djangoapps   | grades                           | 111             | Removed unused `_iter_scorable_xmodules` from `transformer.py` file.                                                                                                                                                                                               |
| lms         | djangoapps   | instructor                       | 294             | Didn't rename `module_state_key`, as it's `StudentModule`'s field, didn't rename `delete_module`, as it's an API part.                                                                         |
| lms         | djangoapps   | lms\_initialization              | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | survey                           | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | experiments                      | 6               |                                                                                                                                                                                                |
| lms         | djangoapps   | certificates                     | 9               |                                                                                                                                                                                                |
| lms         | djangoapps   | lti\_provider                    | 9               |                                                                                                                                                                                                |
| lms         | djangoapps   | rss\_proxy                       | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | instructor\_analytics            | 11              |                                                                                                                                                                                                |
| lms         | djangoapps   | coursewarehistoryextended        | 52              |                                                                                                                                                                                                |
| lms         | djangoapps   | learner\_home                    | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | commerce                         | 2               |                                                                                                                                                                                                |
| lms         | djangoapps   | course\_api                      | 9               |                                                                                                                                                                                                |
| lms         | djangoapps   | course\_blocks                   | 32              |                                                                                                                                                                                                |
| lms         | djangoapps   | gating                           | 11              |                                                                                                                                                                                                |
| lms         | djangoapps   | save\_for\_later                 | 1               |                                                                                                                                                                                                |
| lms         | djangoapps   | edxnotes                         | 32              |                                                                                                                                                                                                |
| lms         | djangoapps   | course\_wiki                     | 5               |                                                                                                                                                                                                |
| lms         | djangoapps   | monitoring                       | 13              |                                                                                                                                                                                                |
| lms         | djangoapps   | branding                         | 3               |                                                                                                                                                                                                |
| lms         | djangoapps   | badges                           | 8               |                                                                                                                                                                                                |
| lms         | lib          |                                  | 1               |                                                                                                                                                                                                |
| lms         | lib          | courseware\_search               | 1               |                                                                                                                                                                                                |
| lms         | envs         |                                  | 64              |                                                                                                                                                                                                |
| lms         | envs         | docs                             | 9               |                                                                                                                                                                                                |
| pavelib     |              |                                  | 76              |                                                                                                                                                                                                |
| pavelib     | utils        |                                  | 24              |                                                                                                                                                                                                |
| pavelib     | utils        | test                             | 4               |                                                                                                                                                                                                |
| pavelib     | paver\_tests |                                  | 10              |                                                                                                                                                                                                |

## Supporting information

This PR has to be merged simultaneously with:
- https://github.com/openedx/xblock-lti-consumer/pull/320
- https://github.com/mitodl/edx-sga/pull/333

## Testing instructions

Use the sandbox to perform general testing (courseware, course tabs, activation emails, course authoring, course exporting/importing, etc):
- LMS: https://extpr31475.sandbox.opencraft.hosting (credentials: `staff@example.com;edx`)
- Studio: https://studio.extpr31475.sandbox.opencraft.hosting

## Questions

Some `ugettext_lazy`-wrapped strings were changed. Should I re-generate `*.po` files and update translations, or this is done automatically by some regular job, like dependencies upgrades?

Private-ref: [BB-6926](https://tasks.opencraft.com/browse/BB-6926)

0x29a committed:
refactor: rename module -> block within lms/djangoapps/courseware

Also, removed unused `_has_access_xmodule` methid from `lms/djangoapps/courseware/access.py`.

openedx-webhooks commented:
Thanks for the pull request, @0x29a!


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

0x29a committed:
refactor: rename module -> block within lms/djangoapps/instructor_task

0x29a committed:
refactor: rename module -> block within xmodule

0x29a committed:
refactor: rename module -> block within openedx/core

0x29a committed:
refactor: rename module -> block within remaining openedx/features

0x29a committed:
refactor: rename module (or item) -> block within cms

0x29a committed:
refactor: rename module (or item) -> block within common

0x29a committed:
refactor: rename module (or item) -> block within lms/djangoapps/instructor

0x29a committed:
refactor: rename module (or item) -> block within remaining lms

Also, removed `_iter_scorable_xmodules` method from `lms/djangoapps/grades/transformer.py` file.

Agrendalath commented on xmodule/x_module.py:
```diff
@@ -519,7 +519,7 @@ def get_child(self, usage_id):
-    def get_required_module_descriptors(self):
+    def get_required_block_descriptors(self):
         """Returns a list of XModuleDescriptor instances upon which this module depends, but are
         not children of this module"""
```
```suggestion
    def get_required_block_descriptors(self):
        """
        Return a list of XBlock instances upon which this block depends but are
        not children of this block.
        
        TODO: Move this method directly to the ConditionalBlock.
        """
```

This isn't used anywhere else, so let's add a note about it.

Agrendalath commented on xmodule/x_module.py:
```diff
@@ -1656,15 +1656,15 @@ class ModuleSystem(MetricsMixin, ConfigurableFragmentWrapper, ModuleSystemShim,
+        get_block,
         descriptor_runtime,
-        **kwargs,
+        **kwargs
```
```suggestion
        **kwargs,
```

Nit: please keep trailing commas. Even though we're not going to add anything after `**kwargs` (so it doesn't matter in the context of producing a shorter `git diff`), it keeps the code consistent.

Agrendalath commented on lms/djangoapps/courseware/tests/test_block_render.py:
```diff
@@ -238,9 +238,9 @@ def tearDown(self):
         super().tearDown()
 
     def test_get_module(self):
```
```suggestion
    def test_get_block(self):
```

Agrendalath commented on lms/djangoapps/courseware/tests/test_block_render.py:
```diff
@@ -925,7 +925,7 @@ def test_anonymous_user_not_be_graded(self, mock_score_signal):
     @ddt.unpack
-    @patch('lms.djangoapps.courseware.module_render.get_module_for_descriptor', wraps=get_module_for_descriptor)
+    @patch('lms.djangoapps.courseware.block_render.get_block_for_descriptor', wraps=get_block_for_descriptor)
     def test_will_recheck_access_handler_attribute(self, handler, will_recheck_access, mock_get_module):
```
```suggestion
    def test_will_recheck_access_handler_attribute(self, handler, will_recheck_access, mock_get_block):
```

\+ its references in this method.

Agrendalath commented on lms/djangoapps/instructor_task/tests/test_tasks.py:
```diff
@@ -292,7 +292,7 @@ def test_overriding_non_scorable(self):
         with patch(
-                'lms.djangoapps.instructor_task.tasks_helper.module_state.get_module_for_descriptor_internal'
+                'lms.djangoapps.instructor_task.tasks_helper.module_state.get_block_for_descriptor_internal'
         ) as mock_get_module:
```
```suggestion
        ) as mock_get_block:
```

There are eight occurrences of this name in this file.

Agrendalath commented on xmodule/conditional_block.py:
```diff
@@ -317,9 +317,11 @@ def get_required_blocks(self):
         """
-        return [self.system.get_module(descriptor) for descriptor in self.get_required_module_descriptors()]
+        return [
+            self.system.get_block_for_descriptor(descriptor) for descriptor in self.get_required_block_descriptors()
```
Nit: as we're changing these methods, we could replace `system` with `runtime`. `system` is only a backward compatibility alias. We could do the same in `xmodule/split_test_block.py` (and optionally in related tests).

0x29a replied:
@Agrendalath, do you think we can do this everywhere? I did this in the last commit, but can revert and limit the change only to these two files.

Agrendalath replied:
@0x29a, sure, that sounds reasonable to replace them. There are still few remaining usages (e.g., `ag -Q ".system." --py`), but this looks much better. Thanks!

0x29a replied:
Ah, I missed them somehow. Fixed now, @Agrendalath. All remaining ones aren't referencing `XModuleMixin` or XBlock shim's `system`.

Agrendalath commented on common/djangoapps/util/module_utils.py:
```diff
@@ -3,34 +3,34 @@
 """
```
Should we rename this file to `block_utils.py` and change `courseware modules` in this module's docstring to `CourseBlocks`?

0x29a replied:
@Agrendalath, right. Done.

Agrendalath commented on lms/djangoapps/courseware/block_render.py:
```diff
@@ -306,26 +305,26 @@ def get_module(user, request, usage_key, field_data_cache,
-                                         course=course, will_recheck_access=will_recheck_access)
+        return get_block_for_descriptor(user, request, descriptor, field_data_cache, usage_key.course_key,
+                                        position=position,
+                                        wrap_xmodule_display=wrap_xblock_display,
```
We are changing `wrap_xmodule_display` to `wrap_xblock_display`in `get_block`. Is there a specific reason we are not doing the same for `get_block_for_descriptor` and `get_module_system_for_user`?

0x29a replied:
@Agrendalath, no, I guess I was unsure if it's worth to rename this argument at all, and then accidentally missed to revert all related changes. After all, I think it should be renamed. I included the change in the respective commit.

Agrendalath commented on lms/djangoapps/courseware/access.py:
```diff
@@ -597,17 +597,6 @@ def can_load():
     return _dispatch(checkers, action, user, descriptor)
 
 
-def _has_access_xmodule(user, action, xmodule, course_key):
-    """
-    Check if user has access to this xmodule.
-
-    Valid actions:
-      - same as the valid actions for xmodule.descriptor
-    """
-    # Delegate to the descriptor
-    return has_access(user, action, xmodule.descriptor, course_key)
-
-
```
This is no longer used, but please mention that we removed it in the PR's description and the relevant commit's body.

Agrendalath commented on lms/djangoapps/grades/transformer.py:
```diff
@@ -168,18 +168,3 @@ def _collect_grading_policy_hash(cls, block_structure):
             "grading_policy_hash",
             cls.grading_policy_hash(course_block),
         )
-
-    @staticmethod
-    def _iter_scorable_xmodules(block_structure):
-        """
-        Loop through all the blocks locators in the block structure, and
-        retrieve the module (XModule or XBlock) associated with that locator.
-
-        For implementation reasons, we need to pull the max_score from the
-        XModule, even though the data is not user specific.  Here we bind the
-        data to a SystemUser.
-        """
-        for block_locator in block_structure.post_order_traversal():
-            block = block_structure.get_xblock(block_locator)
-            if getattr(block, 'has_score', False):
-                yield block
```
This is no longer used, but please mention that we removed it in the PR's description and the relevant commit's body.

Side note: I didn't add this comment to `lms/djangoapps/instructor/views/tools.py` as that file is much less likely to break anything.

Agrendalath commented on xmodule/modulestore/split_mongo/caching_descriptor_system.py:
```diff
@@ -38,7 +38,7 @@ def __init__(self, modulestore, course_entry, default_class, module_data, lazy,
 
         modulestore: the module store that can be used to retrieve additional
-        modules
+        blocks
```
Tiny nit: let's just merge this with the line above.

Agrendalath commented on xmodule/modulestore/mongo/base.py:
```diff
@@ -1040,22 +1040,22 @@ def get_item(self, usage_key, depth=0, using_descriptor_system=None, for_parent=
             usage_key: a :class:`.UsageKey` instance
-            depth (int): An argument that some module stores may use to prefetch
-                descendents of the queried modules for more efficient results later
+            depth (int): An argument that some block stores may use to prefetch
```
I don't think we want to change `module stores` here.

Agrendalath commented on xmodule/modulestore/split_mongo/split.py:
```diff
@@ -1167,8 +1167,8 @@ def has_item(self, usage_key):
         """
-        depth (int): An argument that some module stores may use to prefetch
-            descendants of the queried modules for more efficient results later
+        depth (int): An argument that some block stores may use to prefetch
```
I don't think we want to change `module stores` here.

Agrendalath commented on xmodule/tests/test_capa_block.py:
```diff
@@ -93,8 +93,8 @@ def create(cls, attempts=None, problem_state=None, correct=False, xml=None, over
-            problem_state: a dict to to be serialized into the instance_state of the
-                module.
+            problem_state: a dict to be serialized into the instance_state of the
+                block.
```
Tiny nit: let's merge this with the line above.

Agrendalath commented on xmodule/seq_block.py:
```diff
@@ -28,7 +28,6 @@
     HTMLSnippet,
     ResourceTemplates,
     shim_xmodule_js,
-    STUDENT_VIEW,
```
Tiny nit: could we remove the one imported from `.x_module` instead? It seems to be a more commonly used pattern to import from `xmodule.x_module`.

0x29a replied:
Yep, done, @Agrendalath.

Agrendalath commented on xmodule/video_block/transcripts_utils.py:
```diff
@@ -556,17 +554,15 @@ def get_or_create_sjson(item, transcripts):
-        generate_sjson_for_all_speeds(item, user_filename, result_subs_dict, item.transcript_language)
-    sjson_transcript = Transcript.asset(item.location, source_subs_id, item.transcript_language).data
+        generate_sjson_for_all_speeds(block, user_filename, result_subs_dict, block.transcript_language)
+        sjson_transcript = Transcript.asset(block.location, source_subs_id, block.transcript_language).data
```
Nice catch 🚀

Agrendalath commented on requirements/edx/base.txt:
```diff
@@ -668,7 +668,7 @@ libsass==0.10.0
 loremipsum==1.0.5
     # via ora2
-lti-consumer-xblock==7.2.0
+-e git+https://github.com/open-craft/xblock-lti-consumer.git@e0b64368bcf0e7d044bf9bf1a4c5ccddc61d49bc#egg=lti-consumer-xblock
```
Just leaving an open comment to remember about changing this:
- [ ] `requirements/edx/base.txt`
- [ ] `requirements/edx/development.txt`
- [ ] `requirements/edx/testing.txt`

Agrendalath reviewed (commented):
@0x29a, as always, fantastic work! I still need to test some parts (I will continue tomorrow), but I'm posting my review to unblock you. Everything looks reasonable to me.

<br/>

> This PR has to be merged simultaneously with: (...) edx-sga

Fortunately, integration tests [are not part of the CI ](https://github.com/openedx/edx-platform/pull/31384#discussion_r1041249388), so this is not a blocker - we can merge https://github.com/mitodl/edx-sga/pull/333 at its own pace.

<br/>

> Some ugettext_lazy-wrapped strings were changed. Should I re-generate *.po files and update translations, or this is done automatically by some regular job, like dependencies upgrades?

These files will be re-generated by the transifex bot. We can remove all translation changes from this PR. Please see [this example change](https://github.com/openedx/edx-platform/pull/30826) and [the follow-up update](https://github.com/openedx/edx-platform/pull/30879).

0x29a committed:
refactor: replace usages of XModuleMixin.system with XBlock.runtime

0x29a commented:
Thank you for the thorough review, @Agrendalath! I think I addressed all comments. Triggering the sandbox update now.

>These files will be re-generated by the transifex bot. We can remove all translation changes from this PR. Please see https://github.com/openedx/edx-platform/pull/30826 and https://github.com/openedx/edx-platform/pull/30879.

Got it, thanks for the explanation. I left file rename changes in `*.po` files, as I'm not sure this script is clever enough to change the path instead of adding new translation records.

Agrendalath reviewed (approved):
:+1:

* [x] I tested this: checked for regressions
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

Agrendalath commented:
@ormsbee, would you like to do a sanity check here too? I will prepare an announcement for this and #31384 once we merge them.

Agrendalath commented:
@ormsbee, just a friendly reminder about this PR.

ormsbee reviewed (approved):
@Agrendalath, @0x29a : This is fantastic. I'm sorry for not getting back to you earlier. For things like this, feel free to merge without blocking on me–just as long as you coordinate with @jristau1984 on the window. Please *do* block on me if you're making a breaking change, model change, or database migration.

Thank you.

Agrendalath committed:
chore: update LTI Consumer XBlock

openedx-webhooks commented:
@0x29a 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31475&entry.1484655318=https://github.com/0x29a&entry.752974735=2022-12-25+12:04&entry.1917517419=2023-01-30+17:49&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->

bradenmacdonald commented:
This makes me so happy :) Thanks for getting this done!
